### DISCUSSION TITLE
Upgrade to Debian 12 (Bookworm)

### DISCUSSION THREAD

**Discussion by GourmetSaint** [telekom-security/tpotce#1347] [Ideas]
With a stable release of Debian 12, will tpot be updated to support this? Update the installer and update scripts to use Debian 12 netinstall? New iso? Thanks in advance.

**Comment by t3chn0m4g3** [comment]
Please see #1325.

**Comment by GourmetSaint** [comment]
Thanks! Worked perfectly!Get Outlook for Android

**Comment by lazywebm** [comment]
FYI I was able to install tpot (standalone) on a freshly deployed vanilla Debian 12 / Bookworm yesterday, with minimal changes to the installer script. Works as intended as far as I can see.

Bonus - I replaced the standard Debian docker packages with docker-ce (currently v24.0.6 vs. old 20.10.24) by setting up the official Docker apt repo before executing the installer.

```
--- a/bin/tpdclean.sh
+++ b/bin/tpdclean.sh
-docker-compose -f /opt/tpot/etc/tpot.yml down -v >> /dev/null 2>&1
-docker-compose -f /opt/tpot/etc/tpot.yml rm -v >> /dev/null 2>&1
+docker compose -f /opt/tpot/etc/tpot.yml down -v >> /dev/null 2>&1
+docker compose -f /opt/tpot/etc/tpot.yml rm -v >> /dev/null 2>&1

--- a/host/etc/systemd/tpot.service
+++ b/host/etc/systemd/tpot.service
-ExecStart=/usr/bin/docker-compose -f /opt/tpot/etc/tpot.yml up --no-color
+ExecStart=/usr/bin/docker compose -f /opt/tpot/etc/tpot.yml up --no-color
 
[...]
 
-ExecStop=/usr/bin/docker-compose -f /opt/tpot/etc/tpot.yml down -v
+ExecStop=/usr/bin/docker compose -f /opt/tpot/etc/tpot.yml down -v


--- a/iso/installer/install.sh
+++ b/iso/installer/install.sh
-myLSB_STABLE_SUPPORTED="bullseye"
+myLSB_STABLE_SUPPORTED="bookworm"

[...]

-$myRANDOM_MINUTE $myPULL_HOUR * * *      root    docker-compose -f /opt/tpot/etc/tpot.yml pull
+$myRANDOM_MINUTE $myPULL_HOUR * * *      root    docker compose -f /opt/tpot/etc/tpot.yml pull

[...]

-pip3 install glances[docker] yq

--- a/packages.txt
+++ b/packages.txt
-docker.io
-docker-compose
+docker-ce
+docker-ce-cli
+docker-compose-plugin
+containerd.io
+glances
+yq
```

**Reply by unmute-me** [reply]
Hello! I have some questions about this. is there a way to talk

**Reply by unmute-me** [reply]
This is supposed to be a bash script right? I tried it but its not working

**Reply by lazywebm** [reply]
No, this is the redacted `git diff` output after applying my changes locally within the tpot repo. You can see which files were updated with the edited lines.
Hint - if you don't set up the docker-ce apt repo like I did, don't update `docker-compose` with `docker compose` or the installer will fail.

**Reply by unmute-me** [reply]
is the "-pip3 install glances[docker] yq" in the installer.sh file because I didnt find it. or do I have to set up the official Docker apt repo before it shows in that file

**Reply by unmute-me** [reply]
nvm found it with:
`grep -n 'pip' install.sh`
